### ISSUE TITLE
[TGL] H264 decode with rendering is using EU instead of Vebox

### ISSUE THREAD

**Issue by VUNNYSODHI** [Intel-Media-SDK/MediaSDK#2568]
**Environment:**
OS: 18.04.5 LTS
libva: 2.8.0
Media-driver: 20.4.4 (104bf9ab)
MSDK: Last master 77f673853c23a9124174b1e4981b0785b03922db 

**Command Used:**
“sample_decode h264 -hw -vaapi -i h264-test_1920x1080.nv12 -r  -rgb4" 

**Behavior:**
When checked with intel-gpu-top tool, we can see for composition its using rendering engine, however from gen11 onwards VEBOX can support CSC NV12 to RGB, it should not use EU.

**Comment by dmitryermilov** [comment]
I don’t think we have libva flags responsible for forcing EUs or VEBOX based processing for CSC. From MSDK point of view it’s black box which engine UMD chooses. Yes, there are VA_FILTER_SCALING_HQ, VA_FILTER_SCALING_FAST, etc but these about scaling. So I would forward the request to UMD side. In an absence of  corresponding API, UMD should choose “the best” (but that is the best? Fastest or with higher VQ or ..) engine.
What do you think, @FurongZhang ?

**Comment by FurongZhang** [comment]
I am trying to find a developer to look into this issue. Yes, currently, no flag to indicate CSC is on VEBOX or SFC, or Render Engine. Driver usually use SFC as the first option.

**Comment by VUNNYSODHI** [comment]
Hi @FurongZhang any update on this issue?

**Comment by FurongZhang** [comment]
https://github.com/Intel-Media-SDK/MediaSDK/pull/2606

**Comment by FurongZhang** [comment]
@VUNNYSODHI

**Comment by VUNNYSODHI** [comment]
Thanks @FurongZhang for the update.

**Comment by dmitryermilov** [comment]
Furong, in our email thread it seems we agreed it's better driver use fixed functions as a default path. However a PR proposes explicit API hand and explicit sample command line. Why is it better?

**Comment by FurongZhang** [comment]
No, @Sodhi, Vunny. Driver’s default mode is SFC. MSDK maps the MSDK default mode to driver EU path, maps the MSDK low power mode to driver Default path(SFC). 

If we change driver or MSDK, it needs to update a lot of validation reference. I don’t prefer to do that. Hence, we’d better to change that in sample code.

**Comment by VUNNYSODHI** [comment]
Hi @FurongZhang, 

As per @dmitryermilov suggestion is better to fix at MSDK level, as we encounter same issue with sample_vpp and sample_multi_transcode, so keeping same changes for each application to force MSDK to use SFC will be tedious effort.

**Comment by dmitryermilov** [comment]
@FurongZhang ,
Can you say what libva flag in MSDK VPP code forces EU path ? We can check together if it's really needed.

**Comment by FurongZhang** [comment]
We have closed this issue.
